### ISSUE TITLE
Document combinator-matchers

### ISSUE THREAD

**Issue by alysbrooks** [lambdaisland/kaocha#228]
Our `combinator-matchers` support is "bring your own," but this is not documented anywhere. Nor do we have much description of this feature or how it works. I suspect it's pretty self-explanatory if you're familiar with `matcher-combinators`, but I think we need to at least point people in the right direction.

**Comment by plexus** [comment]
Good point. IIRC there are two aspects to this, one is dealing with matcher-combinator specific events. I believe this is actually no longer necessary since newer versions no longer emit custom event types, but not 100% sure. I do know there's been discussion about this since every test runner/reporter otherwise needs to be extended to know about matcher-combinators.

The other part is extending `deep-diff` so that the "diffs" (Match/Mismatch) objects that MC emits are printed correctly.

Both is indeed done in a way that does not require loading matcher-combinators on our end, so that we don't force that dependency onto users.

**Comment by alysbrooks** [comment]
This is an important enough documentation gap I think it should remain open.
